src/request.ts

```diff
@@ -1,11 +1,16 @@
-import { RequestInit } from 'node-fetch';
-import * as ProxyAgent from 'proxy-agent';
+import fetch, { RequestInit } from 'node-fetch';
 
-const fetch = require('node-fetch');
+export default async (uri: string, opts: RequestInit = {}) => {
+  let options = opts;
 
-export default (uri: string, opts: RequestInit = {}) => {
-  const options =
-    typeof process === 'object' && process.env.PROXY ? { ...opts, agent: new ProxyAgent(process.env.PROXY) } : opts;
+  if (typeof process === 'object' && process.env.PROXY) {
+    try {
```

Comment:
Is this going to import every time a request is made? Is that gonna be ok for performance/memory usage?

Reply:
Good point!
Luckily, Node.js [caches](https://nodejs.org/api/modules.html#modules_caching) modules (can be checked under [require.cache](https://nodejs.org/api/modules.html#modules_require_cache)), therefore they are loaded and executed only once unless one clears the cache.
That said, we should be okay.

Note, even though `import` is used here, it's replaced with a `require` call when we compile the code from TS to JS.

I don't take bundlers into account, since we don't aim to run this code in a browser, but they are no different.